app/src/main/java/de/s3xy/architecturesample/search/presenter/SearchPresenter.java

```diff
@@ -38,25 +40,46 @@ public void attachView(SearchRepositoriesView view) {
     }
 
     @Override
-    public void dettachView() {
+    public void detachView() {
         mSubscription.unsubscribe();
     }
 
     public void setupSearchListener() {
-        mView.showLoading();
         mSubscription = mView
                 .getQueryTextObservable()
                 .subscribeOn(AndroidSchedulers.mainThread())
                 .debounce(500, TimeUnit.MILLISECONDS)
                 .filter(charSequence -> charSequence.length() > 5)
                 .map(CharSequence::toString)
-                .subscribeOn(Schedulers.io())
-                .concatMap(mGithubApi::searchRepositories)
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnNext(s -> mView.showLoading())
+                .observeOn(Schedulers.io())
+                .concatMap(mInteractor::searchRepositories)
                 .map(RepositoriesSearchResult::getItems)
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(repositories -> {
-                    mView.hideLoading();
-                    mView.showRepositories(repositories);
-                });
+                            mView.hideLoading();
+                            mView.showRepositories(repositories);
+                        },
+                        throwable -> {
+                            mView.hideLoading();
+                            Timber.e(throwable, "Repositories fetching error");
+                            mView.showError(ErrorType.getErrorType(throwable));
+                        },
+                        () -> Timber.i("Repositories fetching complete"));
+    }
+
+    public boolean shouldShowSignIn() {
+        return mInteractor.shouldBeUnauthorized();
+    }
+
+    public void logout() {
```

Comment:
Should login and logout belong to search presenter? 
I think they are better kept in a Seperate Interactor to avoid implementing authentication in every presenter